utils/interfaces/_library/miscs.py

```diff
@@ -109,9 +125,26 @@ def check(self, data):
             self.log_error(f"{data['log_filename']} content should be an array")
             return
 
-        for trace in data["request"]["content"]:
-            for span in trace:
-                if self.rid:
-                    if self.rid == _get_rid_from_span(span):
-                        self.log_debug(f"Found a trace for {self.message}")
-                        self.set_status(True)
+        check_pass = False
+        span_types = []
+        span_count = len(span_types)
+
+        for span in _get_spans_by_rid(self.rid, data):
```

Comment:
Probably a simpler version would do the job : 

```
        for trace in data["request"]["content"]:
            for span in trace:
                if self.rid == _get_rid_from_span(span):
                        self.log_debug(f"Found a trace for {self.message}")
                        if self.span_type is None:  # no need to check for span type
                            self.set_status(True)
                        else: # check for the span type in all spans included in this trace
                            span_types = [span.get("type") for span in trace]
                             if self.span_type in span_types:
                                    self.set_status(True)
                             else
                                    self.log_error(f"Did not found {self.span_type} in {span_types}")
```

Reply:
I adjusted to use this version, thanks for the suggestion.

Reply:
There was a closure bug that I copy pasted, but I've fixed it in the most recent version